dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/validation/DefaultSchemaValidator.java

```diff
@@ -136,6 +137,21 @@ public DefaultSchemaValidator( SchemaService schemaService )
             }
         }
 
+        // TODO: remove it from here and make expressions be embedded objects
```

Comment:
I think this might fit better in a "ValidationRuleObjectBundleHook.java" (which you would have to create); There we can add custom validation for object, like ValidationRule. That way we dont need to add it together with generic code :)